src/common/interceptors/errors.interceptor.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  CallHandler,
+  ExecutionContext,
+  Injectable,
+  NestInterceptor,
+  HttpException,
+  HttpStatus,
+} from '@nestjs/common'
+import { Observable, throwError } from 'rxjs'
+import { catchError } from 'rxjs/operators'
+import { NotFoundError, ValidationError } from '../errors/errors'
+
+@Injectable()
+export class ErrorsInterceptor implements NestInterceptor {
+  intercept(_: ExecutionContext, next: CallHandler): Observable<any> {
+    return next.handle().pipe(
+      catchError((error) => {
+        switch (true) {
+          case error instanceof NotFoundError:
+            return throwError(
```

Comment:
We're always return `throwError` so it worth returning it once and move logic inside

Reply:
Fixed [here](https://github.com/vladislav-shkutovich/NestJS-Todo-API/pull/3/commits/70dca48751556a855dfbe2762d77aee926e79d7e)